Source/NSDate+TimeAgo.swift

```diff
@@ -1,79 +1,49 @@
-    + "/Frameworks/TimeAgo.framework/NSDateTimeAgo.bundle"
-  let bundle = NSBundle(path: bundlePath)
-  return NSLocalizedString(key, tableName: "NSDateTimeAgo", bundle: bundle!, comment: "")
+  return NSLocalizedString(key, comment: "")
 }
 
-func isDateInPast (date: NSDate) {
+func isInTheFuture(date: NSDate) -> Bool {
   if (date.compare(NSDate()) == NSComparisonResult.OrderedDescending) {
-    NSException(
-      name: "NSDate+TimeAgo DateInTheFutureException",
-      reason: "Date the function is being called on has to be in the past or now, but not in the future.",
-      userInfo: nil).raise()
+    return true
   }
+  return false
 }
 
 public extension NSDate {
   var timeAgo: String {
-    isDateInPast(self)
+
+    if isInTheFuture(self) {
+      return NSDateTimeAgoLocalizedStrings("Just now")
+    }
 
     let now = NSDate()
     let seconds = Int(fabs(timeIntervalSinceDate(now)))
     let minutes = Int(round(Float(seconds) / 60.0))
     let hours = Int(round(Float(minutes) / 60.0))
-    let days = Int(round(Float(hours) / 24.0))
-    let weeks = Int(round(Float(days) / 7.0))
-    let months = Int(round(Float(days) / 30.0))
-    let years = Int(round(Float(days) / 365))
-
-    var value: Int!
 
     if seconds < 5 {
-      // Just Now
       return NSDateTimeAgoLocalizedStrings("Just now")
-    } else if seconds < 45 {
-      // Seconds Ago
+    } else if seconds < 60 {
       return stringFromFormat("%%d %@seconds ago", withValue: seconds)
-    } else if seconds < 90 {
-      // A Minute Ago
+    } else if seconds < 120 {
       return NSDateTimeAgoLocalizedStrings("A minute ago")
-    } else if minutes < 45 {
-      // Minutes Ago
+    } else if minutes < 60 {
       return stringFromFormat("%%d %@minutes ago", withValue:minutes)
-    } else if minutes < 90 {
-      // An Hour Ago
+    } else if minutes < 120 {
       return NSDateTimeAgoLocalizedStrings("An hour ago")
     } else if hours < 24 {
-      // Hours Ago
       return stringFromFormat("%%d %@hours ago", withValue: hours)
-    } else if hours < 42 {
-      // Yesterday
-      return NSDateTimeAgoLocalizedStrings("Yesterday")
-    } else if days < 5 {
-      // Days Ago
-      return stringFromFormat("%%d %@days ago", withValue: days)
-    } else if days < 11 {
-      // Last Week
-      return NSDateTimeAgoLocalizedStrings("Last week")
-    } else if days < 30 {
-      // Weeks Ago
-      return stringFromFormat("%%d %@weeks ago", withValue: weeks)
-    } else if days < 45 {
-      // Last month
-      return NSDateTimeAgoLocalizedStrings("Last month")
-    } else if months < 12 {
-      // Month Ago
-      return stringFromFormat("%%d %@months ago", withValue: months)
-    } else if Double(days) < (365 * 1.6) {
-      // Last Year
-      return NSDateTimeAgoLocalizedStrings("Last Year")
+    } else if hours < 24 * 7 {
+      let formatter = NSDateFormatter()
+      formatter.dateFormat = String("EEEE '%@' hh:mm", NSDateTimeAgoLocalizedStrings("at"))
```

Comment:
This may sound really weird in some languages.
It is totally OK to say "Thursday at 15:46", but if you say this in Russian, i.e. "Четверг в 15:46" this would be incorrect.
Need to check - the position of "at" in the sentence as well as the form of the word "day" in Norwegian language.